catalog/app/utils/Data.js

```diff
@@ -28,7 +28,13 @@ const reducer = Action.reducer({
 
 export function useData(request, params, { noAutoFetch = false } = {}) {
   // TODO: accept custom key extraction fn (params => key for comparison)
-  const [state, dispatch] = React.useReducer(reducer, initial)
+  const [state, setState] = React.useState(initial)
+  const stateRef = React.useRef()
+  stateRef.current = state
+
+  const dispatch = (action) => {
+    setState((stateRef.current = reducer(stateRef.current, action)))
```

Comment:
https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/best-practices.js#L266

One day ESLint will catch it as `error`

Reply:
but this is not a "return statement"